### PR TITLE
Update to sodiumoxide 0.2.2

### PR DESCRIPTION
**Pull request by kpp**
@NamsooCho please run tests on your Mac. Remember you added explicit crypto_init() in tests because they fail on your Mac? 2576fcb05a8b15a06

### CODE REVIEW

File: src/toxcore/crypto_core.rs [hunk 1]
```diff
@@ -226,6 +236,32 @@ impl FromBytes for sha512::Digest {
...
     use super::*;
 
+    use std::thread;
+
+    // crypto_init()
+
+    #[test]
+    // test whether crypto init behaves well when one tries to run it in parallel.
+    // Its code should **not** be ran in parallel.
+    fn crypto_init_test() {
```
Comment (kurnevsky) [inline src/toxcore/crypto_core.rs]: Do we need this test? It looks like it should be tested/guaranteed by sodiumoxide.

Comment (kpp) [reply src/toxcore/crypto_core.rs]: No it's not. That feature was removed right before release.

See
 https://github.com/sodiumoxide/sodiumoxide/pull/336
 https://github.com/sodiumoxide/sodiumoxide/blob/master/src/lib.rs#L93

Comment (kurnevsky) [reply src/toxcore/crypto_core.rs]: I know, and that's not what I meant. This test is meant to check thread safety for `crypto_init` but it's not our responsibility to make it thread safe. It should be guaranteed by sodiumoxide,

Comment (kpp) [reply src/toxcore/crypto_core.rs]: I see. You are right

**Pull request merged**